engine/examples/main.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"io"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/googollee/go-socket.io/engine"
+)
+
+func main() {
+	eio := engine.New(
+		engine.OptionPingInterval(25*time.Second),
+		engine.OptionPingTimeout(50*time.Second),
+		engine.OptionMaxBufferSize(1*1024*1024), // 1 MiB
+		engine.OptionDebugLogger(engine.LogDebug, nil),
```

Comment:
i suggest separate turn on debag mode and update logger type
Example:
`engine.OptionDebug()`
`engine.OptionSetLogger()`

Reply:
I hope to combine them in a call. I know they could be set separately, but users should think of them together. What's the meaning of setting debug level but not a logger? Do they forget to set the logger or just want the default one? It could use `nil` as the default logger when putting them together and if users do, it's clear what they want.

Reply:
If the package comes with a predefined logger choice, it should be customizeable with a separate log level option. I agree with separating the log level and logger options.